webapp/app/controllers/test.js

```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  treeCompliantMetadata: function() {
+    var model = this.get('model');
+
+    var metadata = model.get('testMetadata');
+    var compliantObject = [];
+    this.createSubtree(compliantObject, '', metadata);
+    return compliantObject;
+  }.property('model.testMetadata'),
+
+  createSubtree: function myself(array_to_append, parentIndex, treeObject) {
```

Comment:
I agree about the componentization... :)

On Sun, Jan 18, 2015 at 10:59 AM, Rotem Yaari <notifications@github.com>
wrote:

> In webapp/app/controllers/test.js
> <https://github.com/slash-testing/backslash/pull/7#discussion-diff-23131874>
> :
>
> > @@ -0,0 +1,69 @@
> > +import Ember from 'ember';
> > +
> > +export default Ember.Controller.extend({
> > +  treeCompliantMetadata: function() {
> > +    var model = this.get('model');
> > +
> > +    var metadata = model.get('testMetadata');
> > +    var compliantObject = [];
> > +    this.createSubtree(compliantObject, '', metadata);
> > +    return compliantObject;
> > +  }.property('model.testMetadata'),
> > +
> > +  createSubtree: function myself(array_to_append, parentIndex, treeObject) {
>
> Ember has the concept of components, which is intended to avoid UI code in
> controllers
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/slash-testing/backslash/pull/7/files#r23131874>.
>